### ISSUE TITLE
Teleporting 2 players to another world causes the players to become frozen on their client

### ISSUE THREAD

**Issue by AkmalFairuz** [pmmp/PocketMine-MP#5822]
### Issue description
Teleporting 2 players to another world causes the players to become frozen on their client. See https://www.youtube.com/watch?v=NpACT6F8bck

In my opinion, this issue is caused when calling despawnFrom and spawnTo simultaneously (i think this related to https://github.com/pmmp/PocketMine-MP/issues/5810)

### Steps to reproduce the issue
1. Create 2 worlds.
2. Simultaneously teleport 2 players to the other world.
3. Move another player, on your client side the player appears frozen, but actually the player is moving.

### OS and versions

* PocketMine-MP: 4.22.1
* PHP: 8.1

### Plugins
(Type "tp" on the chat to teleport all players to another world)
```php
<?php

declare(strict_types=1);

namespace bug;

use pocketmine\event\Listener;
use pocketmine\event\player\PlayerChatEvent;
use pocketmine\plugin\PluginBase;
use pocketmine\Server;

class Main extends PluginBase implements Listener{

    const DEFAULT_WORLD_NAME = "world";
    const TARGET_WORLD_NAME = "target";

    protected function onEnable(): void{
        $this->getServer()->getPluginManager()->registerEvents($this, $this);
        Server::getInstance()->getWorldManager()->loadWorld(self::DEFAULT_WORLD_NAME);
        Server::getInstance()->getWorldManager()->loadWorld(self::TARGET_WORLD_NAME);
    }

    /**
     * @param PlayerChatEvent $event
     * @priority NORMAL
     */
    public function onChat(PlayerChatEvent $event): void{
        if($event->getMessage() === "tp") {
            foreach(Server::getInstance()->getOnlinePlayers() as $player) {
                $player->teleport(Server::getInstance()->getWorldManager()->getWorldByName($player->getWorld()->getFolderName() === self::TARGET_WORLD_NAME ? self::DEFAULT_WORLD_NAME : self::TARGET_WORLD_NAME)->getSpawnLocation());
                $event->getPlayer()->sendMessage("Teleported {$player->getName()}!");
            }
            $event->cancel();
        }
    }
}
```

**Comment by dktapps** [comment]
Yes, it would make sense that the same underlying client bug is responsible.

**Comment by dktapps** [comment]
Sadly I don't see any clear solution for this problem. If anyone in the community has ideas how to fix this, I'd love to hear them.

**Comment by TristanKanner** [comment]
I can’t play on my favorite server because of this bug, so this means a lot to mean if it’s fixed ASAP. So please make this a high priority

**Comment by dktapps** [comment]
Word is that this will be fixed in 1.20.10. Since there is no practical workaround for this on the server side, I'm going to close this.

**Comment by TristanKanner** [comment]
I don’t think 1.20.10 will be released as 1.20.1 was and now there is a better for 1.20.20 (an update after 1.20.10)

If you have words from Mojang developers that contradicts with my point then please tell me the estimated release date of this update.

**Comment by TristanKanner** [comment]
> I don’t think 1.20.10 will be released as 1.20.1 was and now there is a better for 1.20.20 (an update after 1.20.10)
> 
> If you have words from Mojang developers that contradicts with my point then please tell me the estimated release date of this update.

If it’s not confirmed by Mojang please take my word and re-open this and attempt to fix it, this bug has a big impact on my life

**Comment by dktapps** [comment]
> I don’t think 1.20.10 will be released as 1.20.1 was and now there is a better for 1.20.20 (an update after 1.20.10)
> 
> If you have words from Mojang developers that contradicts with my point then please tell me the estimated release date of this update.

Sure, why don't I break my NDA to convince a random nobody that his unsubstantiated assertions are complete nonsense?
